app/lib/features/activities/pages/activities_page.dart

```diff
@@ -1,12 +1,141 @@
+      if (notification.isActerSpace()) {
+        final space = notification.space()!;
+        avatar = Consumer(
+          builder: (context, ref, child) {
+            final spaceProfile = ref.watch(spaceProfileDataProvider(space));
+            return spaceProfile.when(
+              data: (profile) => InkWell(
+                onTap: () => context.goNamed(
+                  Routes.space.name,
+                  pathParameters: {'spaceIdOrAlias': roomId},
+                ),
+                child: ActerAvatar(
+                  mode: DisplayMode.Space,
+                  displayName: profile.displayName,
+                  uniqueId: roomId,
+                  avatar: profile.getAvatarImage(),
+                  size: 48,
+                ),
+              ),
+              error: (error, stackTrace) => Text(
+                  'Failed to load space due to $error'), // FIXME: fallback would be nice
+              loading: () => const Center(child: CircularProgressIndicator()),
+            );
+          },
+        );
+        room = Consumer(
+          builder: (context, ref, child) {
+            final spaceProfile = ref.watch(spaceProfileDataProvider(space));
+            return spaceProfile.when(
+              data: (value) => Text(value.displayName ?? roomId),
+              error: (error, stackTrace) =>
+                  Text('Failed to load space Text(roomId)due to $error'),
+              loading: () => Text(roomId),
+            );
+          },
+        );
+      } else {
+        final convo = notification.convo()!;
+        avatar = Consumer(
+          builder: (context, ref, child) {
+            final profile = ref.watch(chatProfileDataProvider(convo));
+            return profile.when(
+              data: (profile) => InkWell(
+                onTap: () => context.goNamed(
+                  Routes.chatroom.name,
+                  pathParameters: {'roomId': roomId},
+                  extra: convo,
+                ),
+                child: ActerAvatar(
+                  mode: DisplayMode.GroupChat,
+                  displayName: profile.displayName,
+                  uniqueId: roomId,
+                  avatar: profile.getAvatarImage(),
+                  size: 48,
+                ),
+              ),
+              error: (error, stackTrace) =>
+                  Text('Failed to load room due to $error'),
+              loading: () => const Center(child: CircularProgressIndicator()),
+            );
+          },
+        );
+        room = Consumer(
+          builder: (context, ref, child) {
+            final profile = ref.watch(chatProfileDataProvider(convo));
+            return profile.when(
+              data: (value) => Text(value.displayName ?? roomId),
+              error: (error, stackTrace) =>
+                  Text('Failed to load room due to $error'),
+              loading: () => Text(roomId),
+            );
+          },
+        );
+      }
+    } else {
+      avatar = null;
```

Comment:
This would have a potential of throwing exception in widget building as it can't render ```NULL```. Make it ```const SizedBox.shrink()```

Reply:
but `Null` is a valid `child`-value for `leading: SizedBox(height: 50, width: 50, child: avatar),`. I'd rather keep the tree smaller and use `null` when things are `null` (and `null` is supported).

Reply:
I'm not sure reaching this state is very low probability, and to avoid UI exceptions for ```build()``` as what it would render in ```NULL``` case ?

Reply:
If this is tested, then we can safely ignore it 👍🏻